spec/lib/slack_adapter_spec.rb

```diff
@@ -0,0 +1,115 @@
+require 'rails_helper'
+
+describe SlackAdapter do
+  subject { SlackAdapter.new }
+
+  describe "#channels_create" do
+    it "is a thin wrapper around Slack Gem's method" do
+      slack_double = double(channels_create: true)
+      stub_slack!(slack_double)
+      expect(slack_double).to receive(:channels_create)
+      subject.channels_create
+    end
+  end
+
+  describe "#send_team_invite" do
+    before do
+      allow(Time).to receive(:now).and_return(42)
+    end
+
+    it "executes a post with the right queries" do
+      email = "some_email"
+      payload = "_attempts=1&channels=C067NNHEY&email=#{email}&set_active=true&team_id=#{ENV['SLACK_TEAM_ID']}&token=#{ENV['SLACK_TEST_TOKEN']}"
+      proper_args = { method: :post, url: "https://datamarathontest.slack.com/api/users.admin.invite?t=42", payload: payload }
+      expect(RestClient::Request).to receive(:execute).with(proper_args).and_return({ok: true}.to_json)
+      subject.send_team_invite(email)
+    end
+
+    it "returns proper error if applicable" do
+      allow(RestClient::Request).to receive(:execute).and_return("{\"error\":\"already_in_team\"}")
+      expect{subject.send_team_invite("email")}.to raise_error(SlackAdapter::SlackTeamInviteError)
+    end
+  end
+
+  describe "#send_channel_invite" do
+    it "is a thin wrapper around Slack Gem's method" do
+      slack_double = double(channels_invite: { "ok" => true })
+      stub_slack!(slack_double)
+      expect(slack_double).to receive(:channels_invite)
+      subject.send_channel_invite('channel_id', 'user_id')
+    end
+
+    it "returns true if successful" do
+      slack_double = double(channels_invite: { "ok" => true })
+      stub_slack!(slack_double)
+      expect(subject.send_channel_invite('channel_id', 'user_id')).to eq true
+    end
+
+    it "raises an error if user is already in channel" do
+      slack_double = double(channels_invite: { "ok" => false })
+      stub_slack!(slack_double)
+      expect{subject.send_channel_invite('channel_id', 'user_id')}.to raise_error(SlackAdapter::SlackTeamInviteError)
```

Comment:
Line is too long. [117/80]<br>Space missing to the left of {.<br>Space missing inside {.<br>Space missing inside }.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.